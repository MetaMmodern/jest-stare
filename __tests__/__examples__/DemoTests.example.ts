const fs = require('fs');
const path = require('path');

describe("Demonstrate a jest-stare report", () => {
    it("should be pending", () => {
        pending();
    });

    it("should pass", () => {
        // do nothing
    });

    it("should fail with mismatched snapshot", () => {
        expect({ firstKey: "firstValue", secondKey: "wrongValue" }).toMatchSnapshot();
    });

    it("should pass with snapshot", () => {
        expect({firstKey: "firstValue", secondKey: "secondValue"}).toMatchSnapshot();
    });

    it("should add a snapshot", () => {
        expect("randomString").toMatchSnapshot();
    });

    it("should pass with image snapshot", () => {
        const referenceImage = path.resolve(__dirname, 'reference_images', 'A.png');
        const imageAtTest = fs.readFileSync(referenceImage);

        expect(imageAtTest).toMatchImageSnapshot();
    });

    it("should fail with image snapshot", () => {
        //Stored snapshot is for A.png, so should fail
        const referenceImage = path.resolve(__dirname, 'reference_images', 'B.png');
        const imageAtTest = fs.readFileSync(referenceImage);

        expect(imageAtTest).toMatchImageSnapshot();
    });
});
