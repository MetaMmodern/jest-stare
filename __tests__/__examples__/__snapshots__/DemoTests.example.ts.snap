// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demonstrate a jest-stare report should add a snapshot 1`] = `"randomString"`;

exports[`Demonstrate a jest-stare report should fail with mismatched snapshot 1`] = `
Object {
  "firstKey": "firstValue",
  "secondKey": "secondValue",
}
`;

exports[`Demonstrate a jest-stare report should pass with snapshot 1`] = `
Object {
  "firstKey": "firstValue",
  "secondKey": "secondValue",
}
`;
